test/with-utility.connect.test.js

```diff
@@ -0,0 +1,156 @@
+    const { loading, error, open } = this.props.utilityConnect;
+    return (
+      <div>
+        {loading && <div>Loading...</div>}
+        {error && <div role="alert">{error.message}</div>}
+        <button type="button" disabled={loading} onClick={() => open(config)}>
+          {this.props.buttonLabel}
+        </button>
+      </div>
+    );
+  }
+}
+
+const MockCredentialComponentWithHOC = withUtilityConnect(
+  MockCredentialComponentClass
+);
+
+describe('withUtilityConnect', () => {
+  let props;
+  beforeEach(() => {
+    useScript.mockImplementation(generateUseScriptMock());
+    props = { buttonLabel: 'Click me!' };
+  });
+
+  afterEach(() => {
+    delete global.window._ArcadiaUtilityConnect;
+  });
+
+  it("can forward props to the component it's wrapping", () => {
+    render(<MockCredentialComponentWithHOC {...props} />);
+    expect(
+      screen.getByRole('button', { name: 'Click me!' })
+    ).toBeInTheDocument();
+  });
+
+  it('properly passes loading and error on initialization', () => {
+    render(<MockCredentialComponentWithHOC {...props} />);
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument();
+  });
+
+  it('passes an error if the script fails to load', async () => {
+    useScript.mockImplementation(
+      generateUseScriptMock({ throwScriptError: true })
+    );
+    render(<MockCredentialComponentWithHOC {...props} />);
+    const alert = await screen.findByRole('alert');
+    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+    expect(alert).toHaveTextContent(/Error fetching script/);
+  });
+
+  it('returns the initialization error if the script loads but has trouble initializing', async () => {
+    useScript.mockImplementation(
+      generateUseScriptMock({ loadArcadiaElementSuccessfully: false })
+    );
+    render(<MockCredentialComponentWithHOC {...props} />);
+    const alert = await screen.findByRole('alert');
+    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+    expect(alert).toHaveTextContent(
+      /Error loading Arcadia utility connect service/
+    );
+  });
+
+  describe('calling open', () => {
+    let mockConfig;
+    beforeEach(() => {
+      mockConfig = generateSampleConfig();
+    });
+
+    const clickOpen = async () => {
+      await waitFor(() => expect(screen.getByRole('button')).toBeEnabled());
+      userEvent.click(screen.getByRole('button', { name: 'Click me!' }));
+      await waitForElementToBeRemoved(screen.getByText('Loading...'));
+    };
+
+    it('can call open', async () => {
+      render(<MockCredentialComponentWithHOC {...props} />);
+      await clickOpen();
+      expect(window._ArcadiaUtilityConnect.create).toHaveBeenCalledWith(
+        expect.objectContaining(mockConfig)
+      );
+    });
+
+    it("validates the configuration, and, if it's invalid, it throws an error", async () => {
```

Comment:
do we want to be consistent with single/double quotes in these it blocks (looks like most are using single quotes) -- can't remember if we have prettier in this project

Reply:
My prettier changes lines like:
```
'validates the configuration, and, if it\'s invalid, it throws an error'
```
to 
```
"validates the configuration, and, if it's invalid, it throws an error"
```
under the theory that double quotes are preferable to escapes.

Check out this [avoid escape rule](https://eslint.org/docs/rules/quotes#options)

Reply:
lol Im tired of course :)